scipy/spatial/transform/_rotation.pyx

```diff
@@ -367,13 +372,14 @@ cdef double[:, :] _compute_euler_from_quat(
             c = quat[ind, j] + quat[ind, 3]
             d = quat[ind, k] * sign - quat[ind, i]
 
-        _get_angles(angles[ind], extrinsic, symmetric, sign, pi / 2, a, b, c, d)
+        if _get_angles(angles[ind], extrinsic, symmetric, sign, pi / 2, a, b, c, d):
+            gimbal_warning = True
```

Comment:
Why not just:
```
gimbal_warning = _get_angles(angles[ind], extrinsic, symmetric, sign, pi / 2, a, b, c, d)
```

Reply:
Because it's in a loop. Once it's True, we want it to stay true. Otherwise if the last  angle is not a gimbal lock it won't emit the wanring.

Reply:
Ah of course, makes sense.